storage/api/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteLogMetadataManager.java

```diff
@@ -68,39 +68,23 @@ public interface RemoteLogMetadataManager extends Configurable, Closeable {
 
     /**
      * This method is used to update the {@link RemoteLogSegmentMetadata} asynchronously. Currently, it allows to update with the new
-     * state based on the life cycle of the segment. It can go through the below state transitions.
-     * <p>
-     * <pre>
-     * +---------------------+            +----------------------+
-     * |COPY_SEGMENT_STARTED |-----------&gt;|COPY_SEGMENT_FINISHED |
-     * +-------------------+-+            +--+-------------------+
-     *                     |                 |
-     *                     |                 |
-     *                     v                 v
-     *                  +--+-----------------+-+
-     *                  |DELETE_SEGMENT_STARTED|
-     *                  +-----------+----------+
-     *                              |
-     *                              |
-     *                              v
-     *                  +-----------+-----------+
-     *                  |DELETE_SEGMENT_FINISHED|
-     *                  +-----------------------+
-     * </pre>
-     * <p>
```

Comment:
Why do we remote this diagram?

Reply:
I'm trying to avoid duplication here, so we don't have to maintain the same diagram in two different places that can easily diverge. I'm basically replacing that diagram with a direct link to the the class where the diagram lives. 

If you prefer to keep it, I can revert this change.

Reply:
I think we should keep it. Thanks.